### PR TITLE
Reworking HelloTestServer to support a commandline tool and exrm releases

### PR DESCRIPTION
**Pull request by grundrauschen**
As mentioned in the headline a mix task was added to run the test-server from the command line with matching parameters.
Furthermore the user can build an exrm release and use the configuration file to configure the test-server.
The usage is described in the Readme.md.

### CODE REVIEW

**Comment by hwinkel** [comment]
Hi, coverage has been dropped. should we fix that?

**Comment by grundrauschen** [comment]
Some parts are not really testable, because writing a unit test for a call to start an app using a library might not make that much sense. Never the less I will cover some helper methods tomorrow to get closer to 90% again.

**Comment by grundrauschen** [comment]
I now increased the test coverage.
Never the less I found out, that testing the start-up-script is actually kind of hard (besides helper functions), because it interferes with `mix` starting the application automatically for testing.
Running it like `mix test --no-start` would require to boot and especially stop all processes again manually in the most tests.
In my opinion this would make the test less robust just to test functions with call well tested mix functions.
At the moment it is not possible in Elixir to exclude functions from the test coverage, which do not really make sense to be tested, like it is in many other languages.
Although I am strong believing in testing, I would vote to stick with the 82% for this small project than trying to get to 90% and make the test-suite less sound due to side effects.

File: test/mix/tasks/hellotest_test.exs [hunk 1]
```diff
@@ -0,0 +1,31 @@
+defmodule Mix.Tasks.HellotestTest do
+  use ExUnit.Case, async: true
+
+  setup do
+    :meck.new(Application, [:passthrough])
```
Comment (surik) [inline test/mix/tasks/hellotest_test.exs]: why should we do this?

**Comment by surik** [comment]
Maybe we should squash these to fewer commits?

File: test/mix/tasks/hellotest_test.exs (hunk 1 above)
Comment (grundrauschen) [inline test/mix/tasks/hellotest_test.exs]: I am sorry I missed to remove this block.
I tried to box the changes to the Application-variables, but this does not work using meck for single tests.
I removed it in the last commit.

**Comment by grundrauschen** [comment]
I now squashed the commits together

**Comment by mgumz** [comment]
fine with me. we have other, more urgent things on our plate than to polish this little task until it looks like a golden egg.

**Pull request merged**